mfx/optimize/xopt_scans.py

```diff
@@ -148,6 +153,53 @@ def evaluate(input: dict[str, float]) -> dict[str, float]:
     return Evaluator(function=evaluate)
 
 
+def get_evaluator_yag_2d(
+    yag: str = "dg1",
+    goal: tuple[float, float] | None = None,
+):
+    """
+    Alternate evaluator in 2d space.
+
+    As a default, uses the automatic selection of goal position from the
+    user marker PVs.
+    """
+    yag = yag.lower()
+    if yag not in ("xcs1", "dg1", "dg2", "ip"):
+        raise ValueError("Can only use xcs1, dg1, dg2, ip yags.")
+    devices = init_devices()
+    imager = devices[get_yag_key(yag)]
+    image_device = imager.image1.shaped_image
+    mirror = devices["mr1l4_homs"]
+
+    if goal is None:
+        goal = imager.coords.standard_two_corners_target()
+        print(f"Goal is {goal} from camviewer markers")
+    else:
+        print(f"Goal is {goal} from function input")
+
+    fit = ImageProjectionFit()
+
+    def evaluate(input: dict[str, float]) -> dict[str, float]:
+        print(f"Trying {input['mirror_pitch']}")
+        mirror.pitch.set(input["mirror_pitch"]).wait(timeout=20)
+        image_device.trigger().wait(timeout=10)
+        image = image_device.get()
+        print(f"image shape: {image.shape}")
+        # NOTE/TODO: consider adding an averaging step here before fitting
+        fit_result = fit.fit_image(image)
+        results = {}
+        results["centroid_x"] = fit_result.centroid[0]
+        results["centroid_y"] = fit_result.centroid[1]
+        results["rms_size_x"] = fit_result.rms_size[0]
+        results["rms_size_y"] = fit_result.rms_size[1]
+        results["total_intensity"] = fit_result.total_intensity
+        results["objective"] = np.sqrt((fit_result.centroid[0] - goal[0])**2 + (fit_result.centroid[1] - goal[1]))
```

Comment:
TODO: fix incorrect formula or just switch to a numpy built-in

Reply:
Looks like we need to square the second summand.
```suggestion
        results["objective"] = np.sqrt((fit_result.centroid[0] - goal[0])**2 + (fit_result.centroid[1] - goal[1])**2)
```

Reply:
Apparently the correct numpy built-in way would be to make a vector between the two points and use https://numpy.org/doc/2.2/reference/generated/numpy.linalg.norm.html, I think in this case the code is clearer if I simply add in the missing square.